### PR TITLE
Add responsive font sizes

### PR DESCRIPTION
**Pull request by youknowriad**
Just experimenting with opinionated responsive font sizes.

### CODE REVIEW

File: packages/components/src/font-size-picker/index.js [hunk 1]
```diff
@@ -18,11 +18,32 @@ import RangeControl from '../range-control';
...
 const MAX_FONT_SIZE_DISPLAY = '25px';
 
+/* Responsive value helpers */
+const minFontSize = '12px';
+const valueMatcher = '(?<val>[0-9\\.]*[a-z%]*)';
+const valuePlaceholder = '\\k<val>';
+const minMaxRate = '8vw';
+const slope = '0';
+const space = '\\s*';
+const valueRefex = new RegExp(
```
Comment (ntsekouras) [inline packages/components/src/font-size-picker/index.js:34]: I guess this should be `valueRegex`? 😄

File: packages/components/src/font-size-picker/index.js [hunk 2]
```diff
@@ -18,11 +18,32 @@ import RangeControl from '../range-control';
...
 
 const DEFAULT_FONT_SIZE = 'default';
 const CUSTOM_FONT_SIZE = 'custom';
 const MAX_FONT_SIZE_DISPLAY = '25px';
 
+/* Responsive value helpers */
+const minFontSize = '12px';
+const valueMatcher = '(?<val>[0-9\\.]*[a-z%]*)';
+const valuePlaceholder = '\\k<val>';
+const minMaxRate = '8vw';
```
Comment (mtias) [inline packages/components/src/font-size-picker/index.js:31]: we need to consider which ones we'd want to express as attributes without UI so they can be changed by patterns for finer control

File: packages/components/src/font-size-picker/index.js [hunk 3]
```diff
@@ -18,11 +18,32 @@ import RangeControl from '../range-control';
 import { default as UnitControl, useCustomUnits } from '../unit-control';
 import CustomSelectControl from '../custom-select-control';
 import { VisuallyHidden } from '../visually-hidden';
+import CheckboxControl from '../checkbox-control';
```
Comment (mtias) [inline packages/components/src/font-size-picker/index.js:21]: I'd like to try making this a default for font-sizes above a given threshold and maybe opt-in for em/rem.

File: packages/components/src/font-size-picker/index.js (hunk 2 above)
Comment (youknowriad) [reply packages/components/src/font-size-picker/index.js:31]: When we introduce these attributes, we'll be introducing new things to support (bc) and things we can't revisit. I personally feel it's wiser to do it in the other direction. Meaning build a UI to tweak some of these values but store it like today (custom values), validate that it's the kind of customization we want and then split the attribute.

Alternatively, we should find a way to mark blocks/patterns using this new attribute format as something very experimental that can potentially break in the future.

File: packages/components/src/font-size-picker/index.js (hunk 3 above)
Comment (kjellr) [reply packages/components/src/font-size-picker/index.js:21]: There's a lot of variation depending on device size, but generally I've found that font sizes that are `64px` and above tend to break poorly on small screens. Maybe we use that as a threshold and see how it feels?

File: packages/components/src/font-size-picker/index.js [hunk 4]
```diff
@@ -18,11 +18,32 @@ import RangeControl from '../range-control';
...
+const minMaxRate = '8vw';
+const slope = '0';
+const space = '\\s*';
+const valueRefex = new RegExp(
+	`${ space }clamp\\(${ space }calc\\(${ space }${ minFontSize }${ space }-${ space }\\(${ space }\\(${ space }${ minFontSize }${ space }-${ space }${ valueMatcher }${ space }\\)${ space }\\*${ space }${ slope }${ space }\\)${ space }\\)${ space },${ space }${ valuePlaceholder }${ space },${ space }calc\\(${ space }calc\\(${ space }${ minFontSize }${ space }\\+${ space }${ minMaxRate }${ space }\\)${ space }\\+${ space }\\(${ space }\\(${ space }${ valuePlaceholder }${ space }-${ space }calc\\(${ space }${ minFontSize }${ space }\\+${ space }${ minMaxRate }${ space }\\)${ space }\\)${ space }\\*${ space }${ slope }${ space }\\)${ space }\\)${ space }\\)${ space }`,
+	'g'
+);
+
+function clampValue( value ) {
+	return `clamp( calc( ${ minFontSize } - ( ( ${ minFontSize } - ${ value } ) * ${ slope } ) ) , ${ value } , calc( calc( ${ minFontSize } + ${ minMaxRate } ) + ( ( ${ value } - calc( ${ minFontSize } + ${ minMaxRate } ) ) * ${ slope } ) ) )`;
```
Comment (kjellr) [inline packages/components/src/font-size-picker/index.js:40]: @youknowriad I'm having trouble wrapping my head around each part of these equations. Would you mind annotating this a little more? Or if you can point me to a reference/purpose for the slope value here that'd be helpful too. That's the part I'm getting tripped up on.

Comment (youknowriad) [reply packages/components/src/font-size-picker/index.js:40]: It is based on this exploration https://observablehq.com/@eeeps/modulating-input-font-sizes we'll definitely need more inline documentation about this.

Comment (kjellr) [reply packages/components/src/font-size-picker/index.js:40]: That's really helpful, especially the visualization. Thank you!

File: packages/components/src/font-size-picker/index.js (hunk 3 above)
Comment (mtias) [reply packages/components/src/font-size-picker/index.js:21]: Yes, that'd be a reasonable start, I think.

**Comment by mtias** [comment]
I'd love to pick this back up. Can we add a generic `responsiveFonts` in `settings.typography` that just toggles it globally and no UI?

**Comment by mtias** [comment]
@kjellr @jffng if we can get this in as a theme.json opt-in, it'd be nice to include it first thing in TT2 to test it.

**Comment by exonianp** [comment]
Hi, 

I have applied responsive fonts in the mobile version of a project, where the font size is a function of the size of the title of product.

image

It will be great if you can incorporate it in to your project, also considering the cases of "load more", "infinite scroll" implementations of lists that required additional theme customizations. 

Will be happy to share my plugin

**Pull request closed without merging**